### PR TITLE
Add CreateBucketError, ListMultipartUploads functional test

### PR DESCRIPTION
**Pull request by nesty92**
## Description
Add CreateBucketError, ListMultipartUploads functional test

## Motivation and Context
Increase the coverage of functional test in aws-sdk-go

## How to test this PR?
```sh
$ docker build -t minio/mint . -f Dockerfile.dev
$ docker run -e SERVER_ENDPOINT=play.minio.io:9000 -e ACCESS_KEY=Q3AM3UQ867SPQQA43P2F \
             -e SECRET_KEY=zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG \
             -e ENABLE_HTTPS=1 -e MINT_MODE=full minio/mint:latest
```

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#a6d6c236-7eb2-435a-ab42-07c81019b0a2) |
| mint-gateway-azure.sh | :x: [more...](#ce4f6ba1-08db-4416-bb56-2674801f8f91) |
<a name="a6d6c236-7eb2-435a-ab42-07c81019b0a2"></a>
### 8886-9c87508/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:31349
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 47a9f37bee31:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 22 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 6 seconds
(4/13) Running mc tests ... done in 55 seconds
(5/13) Running minio-go tests ... FAILED in 2 minutes and 46 seconds
{
  "args": {},
  "duration": 61639,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
<a name="ce4f6ba1-08db-4416-bb56-2674801f8f91"></a>
### 8886-9c87508/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:31742
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 27e586972e07:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 10 seconds
{"name": "awscli", "duration": 7699, "function": "make_bucket\n", "status": "FAIL", "error": "Could not connect to the endpoint URL: "http://72.28.97.58:31742/awscli-mint-test-bucket-20894""}

Executed 0 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: mint/run/core/aws-sdk-go/quick-tests.go [hunk 1]
```diff
@@ -609,6 +609,114 @@ func testObjectTagging(s3Client *s3.S3) {
...
+
+	_, errCreating := s3Client.CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(bucketName),
+	})
+	if errCreating == nil {
+		failureLog(function, args, startTime, "", "AWS SDK Go CreateBucket Should Return Error for Existing bucket", err).Fatal()
+		return
+	}
+	// Verify valid error response from server.
+	if errCreating.(s3.RequestFailure).Code() != "BucketAlreadyExists" &&
```
Comment (kannappanr) [inline mint/run/core/aws-sdk-go/quick-tests.go:643]: AWS S3 returns success if the region is `us-east-1`

Comment (harshavardhana) [reply mint/run/core/aws-sdk-go/quick-tests.go:643]: Yes that's why we need to use a different region in this test.

File: mint/run/core/aws-sdk-go/quick-tests.go [hunk 2]
```diff
@@ -609,6 +609,114 @@ func testObjectTagging(s3Client *s3.S3) {
...
+	})
+	if err != nil {
+		failureLog(function, args, startTime, "", "AWS SDK Go CreateBucket Failed", err).Fatal()
+		return
+	}
+	defer cleanup(s3Client, bucketName, "", function, args, startTime, true)
+
+	_, errCreating := s3Client.CreateBucket(&s3.CreateBucketInput{
+		Bucket: aws.String(bucketName),
+	})
```
Comment (nitisht) [inline mint/run/core/aws-sdk-go/quick-tests.go:637]: Not sure, why multiple `CreateBucket` calls here?

Comment (nesty92) [reply mint/run/core/aws-sdk-go/quick-tests.go:637]: @nitisht this is for check the error in creating bucket if this bucket already exists

File: mint/run/core/aws-sdk-go/quick-tests.go (hunk 1 above)
Comment (nesty92) [reply mint/run/core/aws-sdk-go/quick-tests.go:643]: @kannappanr @harshavardhana 
Any suggested region to use in this test ? 
What is the expected behavior in minio for this region, as in AWS?

Comment (harshavardhana) [reply mint/run/core/aws-sdk-go/quick-tests.go:643]: any region which is non us-east-1 will work @Nesty92

Comment (nesty92) [reply mint/run/core/aws-sdk-go/quick-tests.go:643]: @kannappanr @harshavardhana play.minio.io is restricted to us-east-1?. I get an error when I try to change the region for this test 
```sh
"error":"InvalidRegion: Region does not match.\n\tstatus code: 400
```

Comment (harshavardhana) [reply mint/run/core/aws-sdk-go/quick-tests.go:643]: Most probably, let's fix that to allow all regions.

Comment (harshavardhana) [reply mint/run/core/aws-sdk-go/quick-tests.go:643]: You should be able to change this now.

**Comment by nesty92** [comment]
@harshavardhana thanks for the help, I changed the region in testCreateBucketError

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#4fd176a7-270a-45e3-8763-da88ce47a907) |
| mint-gateway-azure.sh | :x: [more...](#44743fe6-beae-487c-a2cd-c4424016821b) |
<a name="44743fe6-beae-487c-a2cd-c4424016821b"></a>
### 8886-b534c8d/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30646
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 09f514c3255d:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... FAILED in 5 minutes and 15 seconds
{"name": "awscli", "duration": 313462, "function": "make_bucket\n", "status": "FAIL", "error": "Connect timeout on endpoint URL: "http://72.28.97.55:30646/awscli-mint-test-bucket-5309""}

Executed 0 out of 13 tests successfully.
```
<a name="4fd176a7-270a-45e3-8763-da88ce47a907"></a>
### 8886-b534c8d/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:31092
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 226805b5f56d:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 2 minutes and 25 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-ruby tests ... done in 7 seconds
(4/13) Running mc tests ... done in 1 minutes and 5 seconds
(5/13) Running minio-go tests ... FAILED in 3 minutes and 15 seconds
{
  "args": {},
  "duration": 85455,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 4 out of 13 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
PTAL @nitisht @kannappanr

**Review by nitisht — APPROVED** [review]

**Pull request merged**